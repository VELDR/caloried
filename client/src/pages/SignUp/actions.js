import {
  NEXT_STEP,
  PREVIOUS_STEP,
  RESET_FORM,
  SET_ACCOUNT,
  SET_ACTIVITY_LEVEL,
  SET_GOAL,
  SET_METRICS,
} from './constants';

export const nextStep = () => ({
  type: NEXT_STEP,
});

export const previousStep = () => ({
  type: PREVIOUS_STEP,
});

export const setGoal = (goal) => ({
  type: SET_GOAL,
  goal,
});

export const setMetrics = (metrics) => ({
  type: SET_METRICS,
  metrics,
});

export const setActivityLevel = (activityLevel) => ({
  type: SET_ACTIVITY_LEVEL,
  activityLevel,
});

export const setAccount = (account) => ({
  type: SET_ACCOUNT,
  account,
});

export const resetForm = () => ({
  type: RESET_FORM,
});
