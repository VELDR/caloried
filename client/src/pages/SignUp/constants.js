export const NEXT_STEP = 'NEXT_STEP';
export const PREVIOUS_STEP = 'PREVIOUS_STEP';
export const SET_GOAL = 'SET_GOAL';
export const SET_METRICS = 'SET_METRICS';
export const SET_ACTIVITY_LEVEL = 'SET_ACTIVITY_LEVEL';
export const SET_ACCOUNT = 'SET_ACCOUNT';
export const RESET_FORM = 'RESET_FORM';
