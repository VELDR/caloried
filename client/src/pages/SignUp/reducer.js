import { produce } from 'immer';
import {
  NEXT_STEP,
  PREVIOUS_STEP,
  RESET_FORM,
  SET_ACCOUNT,
  SET_ACTIVITY_LEVEL,
  SET_GOAL,
  SET_METRICS,
} from './constants';

export const initialState = {
  currentStep: 0,
  goal: null,
  metrics: {},
  activityLevel: null,
  account: {},
};

export const storedKey = ['goal', 'metrics', 'activityLevel', 'account', 'currentStep'];

const signUpReducer = (state = initialState, action) =>
  produce(state, (draft) => {
    switch (action.type) {
      case NEXT_STEP:
        if (state.currentStep < 5) {
          draft.currentStep = state.currentStep + 1;
        }
        break;
      case PREVIOUS_STEP:
        if (draft.currentStep !== 0) {
          draft.currentStep = state.currentStep - 1;
        }
        break;
      case SET_GOAL:
        draft.goal = action.goal;
        break;
      case SET_METRICS:
        draft.metrics = action.metrics;
        break;
      case SET_ACTIVITY_LEVEL:
        draft.activityLevel = action.activityLevel;
        break;
      case SET_ACCOUNT:
        draft.account = action.account;
        break;
      case RESET_FORM:
        return initialState;
    }
  });

export default signUpReducer;
