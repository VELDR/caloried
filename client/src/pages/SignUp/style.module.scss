@import '@styles/base';

.register {
  position: relative;
  width: 100%;
  max-width: 50rem;
  @include flex-column(1rem);
  align-items: center;
  .progressContainer {
    @include flex-items-center;
    justify-content: space-around;
    width: 90%;
    .progress {
      @include flex-center-column(0.25rem);
      &__item {
        @include box(2rem);
        @include flex-center;
        padding: 0.25rem;
        border-radius: 50%;
        border: 1px solid var(--color-primary);
        @include apply-at-md-size {
          @include box(1rem);
        }
        svg {
          color: var(--color-primary);
          @include apply-at-md-size {
            @include box(1rem);
          }
        }
      }
      &__label {
        font-size: $font-size-small;
        color: var(--color-text-secondary);
        letter-spacing: 1px;
        display: none;
        @include apply-at-sm-size {
          display: block;
        }
      }
    }
    .active {
      .progress__item {
        background-color: var(--color-primary);
        svg {
          color: var(--color-bg-primary);
          transform: scale(1.15);
          transition: all 0.5s ease-in-out;
        }
      }
      .progress__label {
        color: var(--color-text-primary);
        font-weight: 500;
      }
    }
  }
}
