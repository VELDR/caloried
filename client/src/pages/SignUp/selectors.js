import { createSelector } from 'reselect';
import { initialState } from './reducer';

const selectSignUpState = (state) => state.signUp || initialState;

export const selectCurrentStep = createSelector(selectSignUpState, (state) => state.currentStep);
export const selectGoal = createSelector(selectSignUpState, (state) => state.goal);
export const selectMetrics = createSelector(selectSignUpState, (state) => state.metrics);
export const selectActivityLevel = createSelector(selectSignUpState, (state) => state.activityLevel);
export const selectAccount = createSelector(selectSignUpState, (state) => state.account);
