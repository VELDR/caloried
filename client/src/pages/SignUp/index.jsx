import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { createStructuredSelector } from 'reselect';
import { connect } from 'react-redux';
import { AccessibilityNew, FitnessCenter, Flag, Person } from '@mui/icons-material';

import FormCard from '@components/forms/FormCard';
import { selectCurrentStep } from './selectors';

import classes from './style.module.scss';

const SignUp = ({ currentStep }) => (
  <div className={classes.register}>
    <div className={classes.progressContainer}>
      <div className={`${classes.progress} ${currentStep === 0 ? classes.active : ''}`}>
        <div className={classes.progress__item}>
          <Flag />
        </div>
        <div className={classes.progress__label}>
          <FormattedMessage id="app_weight_goals" />
        </div>
      </div>
      <div className={`${classes.progress} ${currentStep === 1 ? classes.active : ''}`}>
        <div className={classes.progress__item}>
          <AccessibilityNew />
        </div>
        <div className={classes.progress__label}>
          <FormattedMessage id="app_metrics_title" />
        </div>
      </div>
      <div className={`${classes.progress} ${currentStep === 2 ? classes.active : ''}`}>
        <div className={classes.progress__item}>
          <FitnessCenter />
        </div>
        <div className={classes.progress__label}>
          <FormattedMessage id="app_activity_level" />
        </div>
      </div>
      <div className={`${classes.progress} ${currentStep === 3 ? classes.active : ''}`}>
        <div className={classes.progress__item}>
          <Person />
        </div>
        <div className={classes.progress__label}>
          <FormattedMessage id="app_account_setup" />
        </div>
      </div>
    </div>
    <FormCard />
  </div>
);

SignUp.propTypes = {
  currentStep: PropTypes.number,
};

const mapStateToProps = createStructuredSelector({
  currentStep: selectCurrentStep,
});

export default connect(mapStateToProps)(SignUp);
