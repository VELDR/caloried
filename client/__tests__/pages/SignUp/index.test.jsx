import SignUp from '@pages/SignUp';
import { render } from '@utils/testHelper';

let wrapper;

const mockProps = {
  currentStep: 2,
};

beforeEach(() => {
  wrapper = render(<SignUp {...mockProps} />);
});

describe('SignUp Page', () => {
  it('should render the page correctly', () => {
    const { getByTestId } = wrapper;

    expect(getByTestId('sign-up')).toBeInTheDocument();
    expect(getByTestId('form-card')).toBeInTheDocument();
  });

  it('should match with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
