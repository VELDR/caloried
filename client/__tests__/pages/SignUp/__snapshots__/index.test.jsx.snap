// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Page should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="register"
        data-testid="sign-up"
      >
        <div
          class="progressContainer"
        >
          <div
            class="progress "
          >
            <div
              class="progress__item"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="FlagIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M14.4 6 14 4H5v17h2v-7h5.6l.4 2h7V6z"
                />
              </svg>
            </div>
            <div
              class="progress__label"
            >
              Target Berat Badan
            </div>
          </div>
          <div
            class="progress "
          >
            <div
              class="progress__item"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AccessibilityNewIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20.5 6c-2.61.7-5.67 1-8.5 1s-5.89-.3-8.5-1L3 8c1.86.5 4 .83 6 1v13h2v-6h2v6h2V9c2-.17 4.14-.5 6-1l-.5-2zM12 6c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
                />
              </svg>
            </div>
            <div
              class="progress__label"
            >
              Fisik Tubuh
            </div>
          </div>
          <div
            class="progress "
          >
            <div
              class="progress__item"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="FitnessCenterIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20.57 14.86 22 13.43 20.57 12 17 15.57 8.43 7 12 3.43 10.57 2 9.14 3.43 7.71 2 5.57 4.14 4.14 2.71 2.71 4.14l1.43 1.43L2 7.71l1.43 1.43L2 10.57 3.43 12 7 8.43 15.57 17 12 20.57 13.43 22l1.43-1.43L16.29 22l2.14-2.14 1.43 1.43 1.43-1.43-1.43-1.43L22 16.29z"
                />
              </svg>
            </div>
            <div
              class="progress__label"
            >
              Tingkat Aktivitas
            </div>
          </div>
          <div
            class="progress "
          >
            <div
              class="progress__item"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="PersonIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </div>
            <div
              class="progress__label"
            >
              Pengaturan Akun
            </div>
          </div>
        </div>
        <div
          class="formContainer"
          data-testid="form-card"
        >
          <div
            class="container"
            data-testid="welcome"
          >
            <div
              class="body"
            >
              <div
                class="title"
              >
                Langkah Menuju Hidup Sehat
                ðŸŒ±
              </div>
              <div
                class="subtitle"
              >
                Sebelum memulai, kami ingin mengetahui lebih tentang Anda untuk membuat pengalaman yang disesuaikan dengan tujuan Anda.
                ðŸŽ¯
              </div>
            </div>
            <div
              class="footer"
            >
              <div
                class="footer"
                data-testid="form-footer"
              >
                <button
                  class="button footer__right footer__button"
                  data-testid="primary-button"
                  type="submit"
                >
                  Mulai
                   
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="EastIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m15 5-1.41 1.41L18.17 11H2v2h16.17l-4.59 4.59L15 19l7-7-7-7z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="signIn"
        >
          Sudah punya akun?
           
          <a
            href="/sign-in"
          >
            Masuk
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="register"
      data-testid="sign-up"
    >
      <div
        class="progressContainer"
      >
        <div
          class="progress "
        >
          <div
            class="progress__item"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="FlagIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M14.4 6 14 4H5v17h2v-7h5.6l.4 2h7V6z"
              />
            </svg>
          </div>
          <div
            class="progress__label"
          >
            Target Berat Badan
          </div>
        </div>
        <div
          class="progress "
        >
          <div
            class="progress__item"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="AccessibilityNewIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20.5 6c-2.61.7-5.67 1-8.5 1s-5.89-.3-8.5-1L3 8c1.86.5 4 .83 6 1v13h2v-6h2v6h2V9c2-.17 4.14-.5 6-1l-.5-2zM12 6c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
              />
            </svg>
          </div>
          <div
            class="progress__label"
          >
            Fisik Tubuh
          </div>
        </div>
        <div
          class="progress "
        >
          <div
            class="progress__item"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="FitnessCenterIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20.57 14.86 22 13.43 20.57 12 17 15.57 8.43 7 12 3.43 10.57 2 9.14 3.43 7.71 2 5.57 4.14 4.14 2.71 2.71 4.14l1.43 1.43L2 7.71l1.43 1.43L2 10.57 3.43 12 7 8.43 15.57 17 12 20.57 13.43 22l1.43-1.43L16.29 22l2.14-2.14 1.43 1.43 1.43-1.43-1.43-1.43L22 16.29z"
              />
            </svg>
          </div>
          <div
            class="progress__label"
          >
            Tingkat Aktivitas
          </div>
        </div>
        <div
          class="progress "
        >
          <div
            class="progress__item"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="PersonIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
          <div
            class="progress__label"
          >
            Pengaturan Akun
          </div>
        </div>
      </div>
      <div
        class="formContainer"
        data-testid="form-card"
      >
        <div
          class="container"
          data-testid="welcome"
        >
          <div
            class="body"
          >
            <div
              class="title"
            >
              Langkah Menuju Hidup Sehat
              ðŸŒ±
            </div>
            <div
              class="subtitle"
            >
              Sebelum memulai, kami ingin mengetahui lebih tentang Anda untuk membuat pengalaman yang disesuaikan dengan tujuan Anda.
              ðŸŽ¯
            </div>
          </div>
          <div
            class="footer"
          >
            <div
              class="footer"
              data-testid="form-footer"
            >
              <button
                class="button footer__right footer__button"
                data-testid="primary-button"
                type="submit"
              >
                Mulai
                 
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="EastIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m15 5-1.41 1.41L18.17 11H2v2h16.17l-4.59 4.59L15 19l7-7-7-7z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="signIn"
      >
        Sudah punya akun?
         
        <a
          href="/sign-in"
        >
          Masuk
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
